src/client/mercury/mercury.js

```diff
@@ -166,9 +166,14 @@ var Mercury = SparkBase.extend(
   }),
 
   _attemptConnection: function _attemptConnection() {
+    var socket = this._getNewSocket();
```

Comment:
Yes ``_getNewSocket()`` make it easier to mock and return the socket object in tests.